code-studio/build-commands.js

```diff
@@ -49,10 +49,14 @@ exports.browserify = function (config) {
   var command = (shouldWatch ? 'watchify -v' : 'browserify') +
     (shouldMinify ? '' : ' --debug');
 
+  var reactifyStep = '-t reactify';
+
+  var globifyStep = '-t require-globify';
+
   var factorStep = '';
   if (shouldFactor) {
     factorStep = "-p [ factor-bundle -o '" + (shouldMinify ? 'uglifyjs ' : '') +
-      '>' + buildPath + "`basename $FILE .js`" + extension + "']";
+      '>' + buildPath + "`dirname $(realpath --relative-to " + srcPath + " $FILE)`/`basename -s .js -s .jsx $FILE`" + extension + "']";
```

Comment:
This convoluted change allows us to have build targets in subdirectories that end up in a matching directory structure in the build output.  Example: `src/js/levels/contract_match.jsx` transforms into `build/js/levels/contract_match.js`.

I'm open to a more elegant solution.  Unfortunately, factor-bundle doesn't let you pass multiple command-type output arguments (even though their documentation seems to imply you can).